### ISSUE TITLE
Error with JS

### ISSUE THREAD

**Issue by AlejandroBlackburn** [PokemonGoMap/PokemonGo-Map#169]
Recently I clone the repository and I get this errors:

> map.built.js:227 Uncaught ReferenceError: center_lat is not defined

> map.built.js:1082 Uncaught TypeError: Cannot read property 'setMapTypeId' of undefined

> 19 map.built.js:677 Uncaught TypeError: Cannot read property 'getBounds' of undefined

**Comment by chuyskywalker** [comment]
You get those errors from...where?

Why is anything reading `*.built.js` files? Those are an intermediary step...

**Comment by miaojs** [comment]
Also got these from Chrome's console output. No map displays.

**Comment by dmunozfer** [comment]
Same error here afert `git pull`

Fix in issue #116 comments

    npm install -g grunt
    npm install
    npm run-script build

**Comment by chuyskywalker** [comment]
Can y'all take a screenshot of this? The front end should not be loading any of the `.built.` files.

**Comment by miaojs** [comment]
Sure http://imgur.com/zsMqruQ

**Comment by miaojs** [comment]
@dmunozfer Tried that, doesn't fix the problem for me.

**Comment by miaojs** [comment]
In commit 513a6fc3c97fcc426efc518975fe547ca527297f, center_lat/lng is changed to centerLat/Lng. Wonder if this is why.

**Comment by chuyskywalker** [comment]
I genuinely don't know what to do here. `map.built.js` is not a file that we load. Ever. I have no idea what your browser is picking that up. I can't 100% rule it out as a problem with the project though, so I'll leave this open hoping someone else might have an idea.

**Comment by AlejandroBlackburn** [comment]
I decided to return to the previous version and the problems were solved. Recently I returned to try the latest commit and still coming out the same in the console.

**Comment by tkSimon** [comment]
People get `<blah>.built.js` because most moder browsers have javascript source mapping enabled by default and there seems to be something wrong with how the source maps are generated in the grunt file...  it shows `built` for me as well. if i disable js source maps in the dev tools, it says `min` ;)

can't help with that though, i'm a gulp guy

**Comment by tkSimon** [comment]
Created #229 regarding source maps issue

**Comment by chuyskywalker** [comment]
@Chlodochar can you take a look at this one? Looks like the lint auto-formatting may have broken something here?

**Comment by Chlodochar** [comment]
Run `grunt build` and disable your cache.

`center_lat` shouldn't be anywhere in the code anymore. Check the code for yourself. It's `centerLat` now. It's probably a caching problem (HTML is the new version with `centerLat`, but JS is still old version with `center_lat`). Maybe we should do something about caching in Flask.

**Comment by miaojs** [comment]
Not that I doubted you, but grep for `center_lat` indeed returns nothing.

It looks like for me it was indeed caching. Cloudflare was caching the js files. When I connect to the instance directly, map loads.

**Comment by Chlodochar** [comment]
@miaojs any idea if this can be circumvented from Pokemon Go Map? Is it missing some specific caching headers that cloud flare uses?

**Comment by chuyskywalker** [comment]
Most systems employ some kind of cache buster value, like `url/app.min.js?ts-from-last-build=123456`. Flask probably has something like that...

**Comment by miaojs** [comment]
https://support.cloudflare.com/hc/en-us/articles/202775670-How-Do-I-Tell-CloudFlare-What-to-Cache-

tl;dr: Cloudflare respects Cache-control header. 

@chuyskywalker's solution is probably better since requests between builds can come crom cache

**Comment by chuyskywalker** [comment]
I'm working on something, hold tight

**Comment by chuyskywalker** [comment]
Ta da!
